daphne/src/roles.rs

```diff
@@ -55,18 +55,29 @@ where
     fn get_global_config(&self) -> &DapGlobalConfig;
 
     /// Look up the DAP task configuration for the given task ID.
-    async fn get_task_config_considering_taskprov(
+    ///
+    /// If a `report` has been provided, then look for the draft-wang-ppm-dap-taskprov-<nn> extension
+    /// in the report.  If a taskprov task configuration is successfully read from the report,
+    /// the `opt_in_callback` will be called, and if it returns Ok(true) the server will opt-in to the task.
+    /// if it returns Ok(false) or an error then the server will opt-out or return an appropriate error.
+    async fn get_task_config_considering_taskprov<F>(
         &'srv self,
+        version: DapVersion,
         task_id: Cow<'req, Id>,
         report: Option<&ReportMetadata>,
-    ) -> Result<Option<Self::WrappedDapTaskConfig>, DapError>;
+        opt_in_callback: F,
```

Comment:
When we do eventually want to validate the task config (and "opt-out" if we don't like it), where do you imagine this callback coming from? I wonder if it would make more sense to define this as a method on the DapAggregator trait.

Reply:
I considered doing that but went with the callback since the issue called for it, and I wasn't sure if I was missing something or not where the more general callback concept would be useful.  For the code as it is, I think a method on the trait would do, as presumably the Leader DapAggregator would have the comprehensive check and the Helper would just be OK with it.  Happy to do it that way if you don't think we need the generality later.

Reply:
I think the generality will be useful. Besides, as of now the callback is hard-coded: To make it more programmable, we'll eventually have to pass the callback through the functions that use it. It would fit more clearly into the current architecture to have this simply be a trait method.